Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataStore.mm

```diff
@@ -1447,6 +1449,37 @@ - (void)_fetchDataOfTypes:(NSSet<NSString *> *)dataTypes completionHandler:(WK_S
     completionHandler(nil);
 }
 
+- (void)_restoreData:(NSData *)data completionHandler:(void(^)(void))completionHandler
+{
+    WTF::Persistence::Decoder decoder(span(data));
+
+    std::optional<unsigned> currentLocalStorageSerializationVersion;
+    decoder >> currentLocalStorageSerializationVersion;
+    if (!currentLocalStorageSerializationVersion) {
+        completionHandler();
+        return;
+    }
+
+    std::optional<HashSet<String>> dataTypes;
+    decoder >> dataTypes;
+    if (!dataTypes || dataTypes->isEmpty()) {
+        completionHandler();
+        return;
+    }
+
+    if (dataTypes->contains(WKWebsiteDataTypeLocalStorage)) {
```

Comment:
This serialization / deserialization format is not right. If we supported more than 1 type, this wouldn't work.

Upon decoding, you wouldn't know the decoding order of different types.

Reply:
Asked about this on Slack.